misk-redis/src/main/kotlin/misk/redis/FakeRedis.kt

```diff
@@ -71,4 +71,14 @@ class FakeRedis : Redis {
       expiryInstant = clock.instant().plusSeconds(expiryDuration.seconds)
     )
   }
+
+  override fun setnx(key: String, value: ByteString) {
+    // If the key already exists don't overwrite it
+    if (get(key) != null) return
+    // Set the key to expire at the latest possible instant
+    keyValueStore[key] = Value(
```

Comment:
nit: you can use `keyValueStore .putIfAbsent`

Reply:
done